src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs

```diff
@@ -254,12 +254,13 @@ public static int Main()
 }
 ";
             ParseAndValidate(test,
-    // (7,22): error CS1001: Identifier expected
-    //         foreach (int in myarray)   // CS0230
-    Diagnostic(ErrorCode.ERR_IdentifierExpected, "in"),
-    // (7,22): error CS0230: Type and identifier are both required in a foreach statement
-    //         foreach (int in myarray)   // CS0230
-    Diagnostic(ErrorCode.ERR_BadForeachDecl, "in"));
+                // (7,18): error CS1525: Invalid expression term 'int'
```

Comment:
Because in the absence of a simple declaration before the `in`, we parse it as an expression, such as a tuple lvalue or a tuple declaration expression.

---
In reply to: [90170673](https://github.com/dotnet/roslyn/pull/15548#discussion_r90170673) [](ancestors = 90170673)